machine/v1/types_nutanixprovider.go

```diff
@@ -56,6 +56,22 @@ type NutanixMachineProviderConfig struct {
 	// +kubebuilder:validation:Required
 	SystemDiskSize resource.Quantity `json:"systemDiskSize"`
 
+	// bootType indicates the boot type (SECURE_BOOT, UEFI or LEGACY) the VM should use to boot.
+	// If this field is empty, the default boot type LEGACY is used for the VM to boot.
+	// SECURE_BOOT depends on UEFI boot, i.e., enabling SECURE_BOOT means that UEFI boot is also enabled.
+	// +kubebuilder:validation:Enum="";LEGACY;UEFI;SECURE_BOOT
+	// +optional
+	BootType NutanixBootType `json:"bootType,omitempty"`
+
+	// Project optionally identifies a Prism Central project for the Machine's VM to associate with.
+	// +optional
+	Project NutanixResourceIdentifier `json:"project,omitempty"`
+
+	// Categories optionally defines one or more categories (each with name and value) for
+	// the Machine's VM to associate with.
+	// +optional
+	Categories map[string]string `json:"categories,omitempty"`
```

Comment:
We don't use maps within Kube APIs, instead you need to create a struct and use a list type

```
type NutanixCategory strict {
  // +kubebuilder:validation:Required
  Name string `json:"name"`

   // +kubebuilder:validation:Required
  Value string `json:"value"`
}
```
```
// categories ...
// +listType=map
// +listMapKey=name
// +optional
Categories []NutanixCategory `json:"categories"`
```
The map type and key will ensure that each entry in the list has a unique name